### PR TITLE
raftstore: calculate slow score periodically

### PR DESCRIPTION
**Pull request by 5kbpers**
### What problem does this PR solve?

Problem Summary:

Address #10539.

### What is changed and how it works?

What's Changed:

* Add a tick to pd worker, inspect raftstore latency and calculate slow score periodically.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
Introduce a mechanism to detect slow tikv store
```

### CODE REVIEW

File: components/raftstore/src/store/metrics.rs [hunk 1]
```diff
@@ -620,4 +620,15 @@ lazy_static! {
         "Number of peers in hibernated state.",
         &["state"],
     ).unwrap();
+
+    pub static ref STORE_INSPECT_DURTION_HISTOGRAM: HistogramVec =
+        register_histogram_vec!(
+            "tikv_raftstore_slow_score_duration_seconds",
```
Comment (hicqu) [inline components/raftstore/src/store/metrics.rs:626]: slow_score?

File: components/raftstore/src/store/worker/pd.rs [hunk 2]
```diff
@@ -1346,6 +1447,73 @@ where
...
+impl<EK, ER, T> RunnableWithTimer for Runner<EK, ER, T>
+where
+    EK: KvEngine,
+    ER: RaftEngine,
+    T: PdClient + 'static,
+{
+    fn on_timeout(&mut self) {
+        if !self.slow_score.last_tick_finished {
+            self.slow_score
+                .record(self.slow_score.last_tick_id, Duration::from_secs(1));
```
Comment (hicqu) [inline components/raftstore/src/store/worker/pd.rs:1459]: Can `1s` be a constant?

**Comment by hicqu** [comment]
LGTM.

**Review by hicqu — APPROVED** [review]

File: components/raftstore/src/store/worker/pd.rs [hunk 3]
```diff
@@ -496,6 +507,89 @@ fn hotspot_query_num_report_threshold() -> u64 {
     HOTSPOT_QUERY_RATE_THRESHOLD * 10
 }
 
+struct SlowScore {
```
Comment (hunterlxt) [inline components/raftstore/src/store/worker/pd.rs:510]: IMO it's better to place a description of the motivation and calculation method of slow score

**Review by hunterlxt — APPROVED** [review]

**Pull request merged**